lib/mappings/multisigFollowUp.js

```diff
@@ -0,0 +1,35 @@
+import { selectMultisigWalletInfo } from './selectors';
+
+// depends on react-router
+function mapStateToProps(state, otherProps) {
+  const multisigWalletInfo = selectMultisigWalletInfo(state);
+  const walletId = otherProps.match.params.wallet;
+  let joinKey, id, token, cosigner;
+  const info = multisigWalletInfo[walletId];
+
+  if (info) {
+    joinKey = info.joinKey;
+    id = info.id;
+    // find current cosigner
+    cosigner = info.cosigners.find(c => c.token);
+    token = cosigner.token;
+  }
+
+  return {
+    joinKey,
+    token,
+    id,
+    ...otherProps,
+  };
+}
+
+// TODO: create clear ms wallet info action
+// to remove sensitive information from the state
```

Comment:
Could also be a nice [middleware](https://bpanel.org/docs/api-middleware.html)?

Reply:
Thats a good idea